src/source/from-shift.js

```diff
@@ -0,0 +1,205 @@
+'use strict';
+
+function makeEager(obj) {
+	obj.type = 'Eager' + obj.type;
+}
+
+function dummyDeclaredScope(type) {
+	return {
+		type,
+		declaredNames: [],
+		hasDirectEval: false
+	};
+}
+
+function parameterScopeAndLength(params) {
+	let length = 0;
+	let isSimpleParameterList = true;
+
+	loop: for (let param of params) {
```

Comment:
Fun fact: I rewrote the code that handles `break label;` in the SpiderMonkey parser, but I had completely forgotten the existence of this feature :)